doc/v5.0/man5/Open-MPI.5
'\" t
.\" Automatically generated by Pandoc 2.14.2
.\"
.TH "Open-MPI" "5" "" "2021-08-30" "Open MPI"
.hy
.SH NAME
.PP
Open MPI - General information
.SH OPEN MPI
.PP
Open MPI is an open source implementation of MPI (message-passing
interface), the industry-standard specification for writing
message-passing programs.
Message passing is a programming model that gives the programmer
explicit control over interprocess communication.
.PP
The MPI specification was developed by the MPI Forum, a group of
software developers, computer vendors, academics, and computer-science
researchers whose goal was to develop a standard for writing
message-passing programs that would be efficient, flexible, and
portable.
.PP
The outcome, known as the MPI Standard, was first published in 1993; its
most recent version (MPI-3.1) was published in June 2015.
Open MPI includes all MPI 3.1-compliant routines.
.PP
For more information about Open MPI, see https://www.open-mpi.org.
.PP
The MPI standards are available at https://www.mpi-forum.org.
.SH MAN PAGE SYNTAX
.PP
Man pages for Open MPI and Open MPI I/O routines are named according to
C syntax, that is, they begin with the prefix \f[C]MPI_\f[R], all in
uppercase, and the first letter following the \f[C]MPI_\f[R] prefix is
also uppercase.
The rest of the letters in the routine are all lowercase, for example,
\f[C]MPI_Comm_get_attr\f[R].
.SH ENVIRONMENT
.PP
To fine-tune your Open MPI environment, you can either use arguments to
the \f[C]mpirun\f[R] or \f[C]mpiexec\f[R] commands, or you can use MCA
parameters.
.PP
For more information on arguments, see the \f[C]mpirun\f[R](1) man page.
.PP
For a complete listing of MCA parameters and their descriptions, issue
the command \f[C]ompi_info --all\f[R].
This information also appears in the FAQ on the Open MPI web site at
https://www.open-mpi.org/faq/?category=tuning#mca-params.
.SH ERRORS
.PP
All MPI routines (except \f[C]MPI_Wtime\f[R] and \f[C]MPI_Wtick\f[R])
return an error value; C routines as the value of the function and
Fortran routines in the last argument.
Before the value is returned, the current MPI error handler is called.
By default, this error handler aborts the MPI job.
The error handler may be changed with \f[C]MPI_Comm_set_errhandler\f[R];
the predefined error handler \f[C]MPI_ERRORS_RETURN\f[R] may be used to
cause error values to be returned.
Note that MPI does not guarantee that an MPI program can continue past
an error.
.PP
For more information on Open MPI error codes, see \f[C]mpi.h\f[R] in the
\f[C]include\f[R] directory.
.PP
Standard error return classes for Open MPI:
.PP
.TS
tab(@);
lw(23.3n) lw(23.3n) lw(23.3n).
T{
Error name
T}@T{
Error value
T}@T{
Description
T}
_
T{
MPI_SUCCESS
T}@T{
0
T}@T{
Successful return code.
T}
T{
MPI_ERR_BUFFER
T}@T{
1
T}@T{
Invalid buffer pointer.
T}
T{
MPI_ERR_COUNT
T}@T{
2
T}@T{
Invalid count argument.
T}
T{
MPI_ERR_TYPE
T}@T{
3
T}@T{
Invalid datatype argument.
T}
T{
MPI_ERR_TAG
T}@T{
4
T}@T{
Invalid tag argument.
T}
T{
MPI_ERR_COMM
T}@T{
5
T}@T{
Invalid communicator.
T}
T{
MPI_ERR_RANK
T}@T{
6
T}@T{
Invalid rank.
T}
T{
MPI_ERR_REQUEST
T}@T{
7
T}@T{
Invalid MPI_Request handle.
T}
T{
MPI_ERR_ROOT
T}@T{
8
T}@T{
Invalid root.
T}
T{
MPI_ERR_GROUP
T}@T{
9
T}@T{
Null group passed to function.
T}
T{
MPI_ERR_OP
T}@T{
10
T}@T{
Invalid operation.
T}
T{
MPI_ERR_TOPOLOGY
T}@T{
11
T}@T{
Invalid topology.
T}
T{
MPI_ERR_DIMS
T}@T{
12
T}@T{
Illegal dimension argument.
T}
T{
MPI_ERR_ARG
T}@T{
13
T}@T{
Invalid argument.
T}
T{
MPI_ERR_UNKNOWN
T}@T{
14
T}@T{
Unknown error.
T}
T{
MPI_ERR_TRUNCATE
T}@T{
15
T}@T{
Message truncated on receive.
T}
T{
MPI_ERR_OTHER
T}@T{
16
T}@T{
Other error; use Error_string.
T}
T{
MPI_ERR_INTERN
T}@T{
17
T}@T{
Internal error code.
T}
T{
MPI_ERR_IN_STATUS
T}@T{
18
T}@T{
Look in status for error value.
T}
T{
MPI_ERR_PENDING
T}@T{
19
T}@T{
Pending request.
T}
T{
MPI_ERR_ACCESS
T}@T{
20
T}@T{
Permission denied.
T}
T{
MPI_ERR_AMODE
T}@T{
21
T}@T{
Unsupported amode passed to open.
T}
T{
MPI_ERR_ASSERT
T}@T{
22
T}@T{
Invalid assert.
T}
T{
MPI_ERR_BAD_FILE
T}@T{
23
T}@T{
Invalid file name (for example, path name too long).
T}
T{
MPI_ERR_BASE
T}@T{
24
T}@T{
Invalid base.
T}
T{
MPI_ERR_CONVERSION
T}@T{
25
T}@T{
An error occurred in a user-supplied data-conversion function.
T}
T{
MPI_ERR_DISP
T}@T{
26
T}@T{
Invalid displacement.
T}
T{
MPI_ERR_DUP_DATAREP
T}@T{
27
T}@T{
Conversion functions could not be registered because a data
representation identifier that was already defined was passed to
MPI_REGISTER_DATAREP.
T}
T{
MPI_ERR_FILE_EXISTS
T}@T{
28
T}@T{
File exists.
T}
T{
MPI_ERR_FILE_IN_USE
T}@T{
29
T}@T{
File operation could not be completed, as the file is currently open by
some process.
T}
T{
MPI_ERR_FILE
T}@T{
30
T}@T{
Invalid file handle.
T}
T{
MPI_ERR_INFO_KEY
T}@T{
31
T}@T{
Illegal info key.
T}
T{
MPI_ERR_INFO_NOKEY
T}@T{
32
T}@T{
No such key.
T}
T{
MPI_ERR_INFO_VALUE
T}@T{
33
T}@T{
Illegal info value.
T}
T{
MPI_ERR_INFO
T}@T{
34
T}@T{
Invalid info object.
T}
T{
MPI_ERR_IO
T}@T{
35
T}@T{
I/O error.
T}
T{
MPI_ERR_KEYVAL
T}@T{
36
T}@T{
Illegal key value.
T}
T{
MPI_ERR_LOCKTYPE
T}@T{
37
T}@T{
Invalid locktype.
T}
T{
MPI_ERR_NAME
T}@T{
38
T}@T{
Name not found.
T}
T{
MPI_ERR_NO_MEM
T}@T{
39
T}@T{
Memory exhausted.
T}
T{
MPI_ERR_NOT_SAME
T}@T{
40
T}@T{
Collective argument not identical on all processes, or collective
routines called in a different order by different processes.
T}
T{
MPI_ERR_NO_SPACE
T}@T{
41
T}@T{
Not enough space.
T}
T{
MPI_ERR_NO_SUCH_FILE
T}@T{
42
T}@T{
File (or directory) does not exist.
T}
T{
MPI_ERR_PORT
T}@T{
43
T}@T{
Invalid port.
T}
T{
MPI_ERR_PROC_ABORTED
T}@T{
74
T}@T{
Operation failed because a remote peer has aborted.
T}
T{
MPI_ERR_QUOTA
T}@T{
44
T}@T{
Quota exceeded.
T}
T{
MPI_ERR_READ_ONLY
T}@T{
45
T}@T{
Read-only file system.
T}
T{
MPI_ERR_RMA_CONFLICT
T}@T{
46
T}@T{
Conflicting accesses to window.
T}
T{
MPI_ERR_RMA_SYNC
T}@T{
47
T}@T{
Erroneous RMA synchronization.
T}
T{
MPI_ERR_SERVICE
T}@T{
48
T}@T{
Invalid publish/unpublish.
T}
T{
MPI_ERR_SIZE
T}@T{
49
T}@T{
Invalid size.
T}
T{
MPI_ERR_SPAWN
T}@T{
50
T}@T{
Error spawning.
T}
T{
MPI_ERR_UNSUPPORTED_DATAREP
T}@T{
51

doc/v5.0/man5/Open-MPI.5.php
<?php
$topdir = "../../..";
$title = "Open-MPI(5) man page (version 5.0.0rc1)";
$meta_desc = "Open MPI v5.0.0rc1 man page: Open-MPI(5)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
   <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<p>
Open MPI - General information
<h2><a name='sect1' href='#toc1'>Open Mpi</a></h2>
<p>
Open MPI is an open source
implementation of MPI (message-passing interface), the industry-standard
specification for writing message-passing programs. Message passing is a
programming model that gives the programmer explicit control over interprocess
communication. <p>
The MPI specification was developed by the MPI Forum, a group
of software developers, computer vendors, academics, and computer-science
researchers whose goal was to develop a standard for writing message-passing
programs that would be efficient, flexible, and portable. <p>
The outcome, known
as the MPI Standard, was first published in 1993; its most recent version
(MPI-3.1) was published in June 2015. Open MPI includes all MPI 3.1-compliant
routines. <p>
For more information about Open MPI, see <i>https://www.open-mpi.org.</i>
<p>
The MPI standards are available at <i>https://www.mpi-forum.org.</i>
<h2><a name='sect2' href='#toc2'>Man Page Syntax</a></h2>
<p>
Man
pages for Open MPI and Open MPI I/O routines are named according to C syntax,
that is, they begin with the prefix C]MPI_R], all in uppercase, and the
first letter following the C]MPI_R] prefix is also uppercase. The rest of
the letters in the routine are all lowercase, for example, C]MPI_Comm_get_attrR].

<h2><a name='sect3' href='#toc3'>Environment</a></h2>
<p>
To fine-tune your Open MPI environment, you can either use arguments
to the C]mpirunR] or C]mpiexecR] commands, or you can use MCA parameters.
<p>
For more information on arguments, see the C]mpirunR](1) man page. <p>
For a
complete listing of MCA parameters and their descriptions, issue the command
C]ompi_info --allR]. This information also appears in the FAQ on the Open
MPI web site at <i>https://www.open-mpi.org/faq/?category=tuning#mca-params.</i>
<h2><a name='sect4' href='#toc4'>Errors</a></h2>
<p>
All
MPI routines (except C]MPI_WtimeR] and C]MPI_WtickR]) return an error value;
C routines as the value of the function and Fortran routines in the last
argument. Before the value is returned, the current MPI error handler is
called. By default, this error handler aborts the MPI job. The error handler
may be changed with C]MPI_Comm_set_errhandlerR]; the predefined error handler
C]MPI_ERRORS_RETURNR] may be used to cause error values to be returned.
Note that MPI does not guarantee that an MPI program can continue past
an error. <p>
For more information on Open MPI error codes, see C]mpi.hR] in
the C]includeR] directory. <p>
Standard error return classes for Open MPI: <p>
<table border='0'>
<tr><td align='left'>Error
name</td></tr>
<tr><td align='left'>Error value</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Description</td></tr>
<tr><td align='left'>T}</td></tr>
<tr> <tr><td align='left'>MPI_SUCCESS </td></tr>
<tr><td align='left'>0</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Successful return code.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_BUFFER
</td></tr>
<tr><td align='left'>1</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid buffer pointer.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_COUNT </td></tr>
<tr><td align='left'>2</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid count argument.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_TYPE
</td></tr>
<tr><td align='left'>3</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid datatype argument.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_TAG </td></tr>
<tr><td align='left'>4</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid tag argument.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_COMM
</td></tr>
<tr><td align='left'>5</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid communicator.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_RANK </td></tr>
<tr><td align='left'>6</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid rank.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_REQUEST
</td></tr>
<tr><td align='left'>7</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid MPI_Request handle.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_ROOT </td></tr>
<tr><td align='left'>8</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid root.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_GROUP
</td></tr>
<tr><td align='left'>9</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Null group passed to function.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_OP </td></tr>
<tr><td align='left'>10</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid operation.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_TOPOLOGY
</td></tr>
<tr><td align='left'>11</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid topology.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_DIMS </td></tr>
<tr><td align='left'>12</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Illegal dimension argument.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_ARG
</td></tr>
<tr><td align='left'>13</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid argument.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_UNKNOWN </td></tr>
<tr><td align='left'>14</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Unknown error.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_TRUNCATE
</td></tr>
<tr><td align='left'>15</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Message truncated on receive.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_OTHER </td></tr>
<tr><td align='left'>16</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Other error; use Error_string.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_INTERN
</td></tr>
<tr><td align='left'>17</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Internal error code.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_IN_STATUS </td></tr>
<tr><td align='left'>18</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Look in status for error
value.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_PENDING </td></tr>
<tr><td align='left'>19</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Pending request.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_ACCESS </td></tr>
<tr><td align='left'>20</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Permission
denied.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_AMODE </td></tr>
<tr><td align='left'>21</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Unsupported amode passed to open.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_ASSERT
</td></tr>
<tr><td align='left'>22</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid assert.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_BAD_FILE </td></tr>
<tr><td align='left'>23</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid file name (for example,
path name too long).</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_BASE </td></tr>
<tr><td align='left'>24</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid base.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_CONVERSION
</td></tr>
<tr><td align='left'>25</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>An error occurred in a user-supplied data-conversion function.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_DISP
</td></tr>
<tr><td align='left'>26</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid displacement.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_DUP_DATAREP </td></tr>
<tr><td align='left'>27</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Conversion functions
could not be registered because a data</td></tr>
<tr><td align='left'>representation identifier that was
already defined was passed to</td></tr>
<tr><td align='left'><a href="../man3/MPI_Register_datarep.3.php">MPI_REGISTER_DATAREP</a>.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_FILE_EXISTS
</td></tr>
<tr><td align='left'>28</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>File exists.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_FILE_IN_USE </td></tr>
<tr><td align='left'>29</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>File operation could not be completed,
as the file is currently open by</td></tr>
<tr><td align='left'>some process.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_FILE </td></tr>
<tr><td align='left'>30</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid
file handle.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_INFO_KEY </td></tr>
<tr><td align='left'>31</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Illegal info key.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_INFO_NOKEY
</td></tr>
<tr><td align='left'>32</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>No such key.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_INFO_VALUE </td></tr>
<tr><td align='left'>33</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Illegal info value.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_INFO
</td></tr>
<tr><td align='left'>34</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid info object.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_IO </td></tr>
<tr><td align='left'>35</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>I/O error.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_KEYVAL </td></tr>
<tr><td align='left'>36</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Illegal
key value.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_LOCKTYPE </td></tr>
<tr><td align='left'>37</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid locktype.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_NAME </td></tr>
<tr><td align='left'>38</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Name
not found.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_NO_MEM </td></tr>
<tr><td align='left'>39</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Memory exhausted.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_NOT_SAME </td></tr>
<tr><td align='left'>40</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Collective
argument not identical on all processes, or collective</td></tr>
<tr><td align='left'>routines called in
a different order by different processes.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_NO_SPACE </td></tr>
<tr><td align='left'>41</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Not enough
space.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_NO_SUCH_FILE </td></tr>
<tr><td align='left'>42</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>File (or directory) does not exist.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_PORT
</td></tr>
<tr><td align='left'>43</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid port.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_PROC_ABORTED </td></tr>
<tr><td align='left'>74</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Operation failed because a remote
peer has aborted.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_QUOTA </td></tr>
<tr><td align='left'>44</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Quota exceeded.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_READ_ONLY </td></tr>
<tr><td align='left'>45</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Read-only
file system.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_RMA_CONFLICT </td></tr>
<tr><td align='left'>46</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Conflicting accesses to window.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_RMA_SYNC
</td></tr>
<tr><td align='left'>47</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Erroneous RMA synchronization.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_SERVICE </td></tr>
<tr><td align='left'>48</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid publish/unpublish.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_SIZE
</td></tr>
<tr><td align='left'>49</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid size.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_SPAWN </td></tr>
<tr><td align='left'>50</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Error spawning.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_UNSUPPORTED_DATAREP
</td></tr>
<tr><td align='left'>51</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Unsupported datarep passed to <a href="../man3/MPI_File_set_view.3.php">MPI_File_set_view</a>.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_UNSUPPORTED_OPERATION
</td></tr>
<tr><td align='left'>52</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Unsupported operation, such as seeking on a file that supports only</td></tr>
<tr><td align='left'>sequential
access.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_WIN </td></tr>
<tr><td align='left'>53</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid window.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_MEMORY </td></tr>
<tr><td align='left'>54</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Out of memory.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_NOT_INITIALIZED
</td></tr>
<tr><td align='left'>55</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Interface not initialized.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_CANNOT_INIT </td></tr>
<tr><td align='left'>56</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Interface not
in the state to be initialized.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_INVALID_INDEX </td></tr>
<tr><td align='left'>57</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>The enumeration
index is invalid.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_INVALID_ITEM </td></tr>
<tr><td align='left'>58</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>The item index queried is
out of range.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_INVALID_HANDLE </td></tr>
<tr><td align='left'>59</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>The handle is invalid.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_OUT_OF_HANDLES
</td></tr>
<tr><td align='left'>60</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>No more handles available.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_OUT_OF_SESSIONS </td></tr>
<tr><td align='left'>61</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>No more sessions
available.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_INVALID_SESSION </td></tr>
<tr><td align='left'>62</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Session argument is not a valid
session.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_CVAR_SET_NOT_NOW </td></tr>
<tr><td align='left'>63</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Variable cannot be set at this
moment.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_CVAR_SET_NEVER </td></tr>
<tr><td align='left'>64</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Variable cannot be set until end of
execution.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_PVAR_NO_STARTSTOP </td></tr>
<tr><td align='left'>65</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Variable cannot be started or
stopped.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_PVAR_NO_WRITE </td></tr>
<tr><td align='left'>66</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Variable cannot be written or reset.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_PVAR_NO_ATOMIC
</td></tr>
<tr><td align='left'>67</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Variable cannot be read and written atomically.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_RMA_RANGE </td></tr>
<tr><td align='left'>68</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Target
memory is not part of the window (in the case of a window created</td></tr>
<tr><td align='left'>with <a href="../man3/MPI_Win_create_dynamic.3.php">MPI_WIN_CREATE_DYNAMIC</a>,
target memory is not attached).</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_RMA_ATTACH </td></tr>
<tr><td align='left'>69</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Memory cannot be
attached (e.g., because of resource exhaustion).</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_RMA_FLAVOR </td></tr>
<tr><td align='left'>70</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Passed
window has the wrong flavor for the called function.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_RMA_SHARED
</td></tr>
<tr><td align='left'>71</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Memory cannot be shared (e.g., some process in the group of the</td></tr>
<tr><td align='left'>specified
communicator cannot expose shared memory).</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_INVALID </td></tr>
<tr><td align='left'>72</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Invalid
use of the interface or bad parameter values(s).</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_T_ERR_INVALID_NAME
</td></tr>
<tr><td align='left'>73</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>The variable or category name is invalid.</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>MPI_ERR_LASTCODE </td></tr>
<tr><td align='left'>93</td></tr>
<tr><td align='left'>T}</td></tr>
<tr><td align='left'>Last
error code.</td></tr>
<tr><td align='left'>T}</td></tr>
</table>

<h2><a name='sect5' href='#toc5'>See Also</a></h2>
<p>
<i><a href="../man3/MPI_T.5.php">MPI_T</a>(5)</i> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Open Mpi</a></li>
<li><a name='toc2' href='#sect2'>Man Page Syntax</a></li>
<li><a name='toc3' href='#sect3'>Environment</a></li>
<li><a name='toc4' href='#sect4'>Errors</a></li>
<li><a name='toc5' href='#sect5'>See Also</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
